src/Functions/extractTimeZoneFromFunctionArguments.cpp

```diff
@@ -16,25 +16,22 @@ namespace ErrorCodes
     extern const int ILLEGAL_TYPE_OF_ARGUMENT;
 }
 
-
-std::string extractTimeZoneNameFromColumn(const IColumn * column, const String & column_name)
+std::string extractTimeZoneNameFromColumn(const IColumn * column, [[maybe_unused]] const String & column_name)
```

Comment:
Argument allow_nonconst_timezone_arguments is no longer used. Could you maybe just remove it (and adjust the callers)?